From 8f0b00e98165962af784344b5e5148e3f15448ff Mon Sep 17 00:00:00 2001
From: "Rifa'i M. Hanif" <hanreev@gmail.com>
Date: Tue, 6 Aug 2019 22:21:18 +0700
Subject: [PATCH 1/1] fix ol/Observable event listener

---
 src/ol/Observable.js | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ol/Observable.js b/src/ol/Observable.js
index 098e284d8..ffbf9cd01 100644
--- a/src/ol/Observable.js
+++ b/src/ol/Observable.js
@@ -51,7 +51,7 @@ class Observable extends EventTarget {
   /**
    * Listen for a certain type of event.
    * @param {string|Array<string>} type The event type or array of event types.
-   * @param {function(?): ?} listener The listener function.
+   * @param {function(?)} listener The listener function.
    * @return {import("./events.js").EventsKey|Array<import("./events.js").EventsKey>} Unique key for the listener. If
    *     called with an array of event types as the first argument, the return
    *     will be an array of keys.
@@ -73,7 +73,7 @@ class Observable extends EventTarget {
   /**
    * Listen once for a certain type of event.
    * @param {string|Array<string>} type The event type or array of event types.
-   * @param {function(?): ?} listener The listener function.
+   * @param {function(?)} listener The listener function.
    * @return {import("./events.js").EventsKey|Array<import("./events.js").EventsKey>} Unique key for the listener. If
    *     called with an array of event types as the first argument, the return
    *     will be an array of keys.
@@ -95,7 +95,7 @@ class Observable extends EventTarget {
   /**
    * Unlisten for a certain type of event.
    * @param {string|Array<string>} type The event type or array of event types.
-   * @param {function(?): ?} listener The listener function.
+   * @param {function(?)} listener The listener function.
    * @api
    */
   un(type, listener) {
-- 
2.17.1

