diff --git a/src/ol/Geolocation.js b/src/ol/Geolocation.js
index 7bfbbfe14..c5ec305d6 100644
--- a/src/ol/Geolocation.js
+++ b/src/ol/Geolocation.js
@@ -34,18 +34,20 @@ const Property = {
  */
 class GeolocationError extends BaseEvent {
   /**
-   * @param {GeolocationPositionError} error error object.
+   * @param {*} error error object.
    */
   constructor(error) {
     super(EventType.ERROR);
 
     /**
      * @type {number}
+     * @api
      */
     this.code = error.code;
 
     /**
      * @type {string}
+     * @api
      */
     this.message = error.message;
   }
@@ -81,7 +83,7 @@ class GeolocationError extends BaseEvent {
  *       window.console.log(geolocation.getPosition());
  *     });
  *
- * @fires module:ol/events/Event~BaseEvent#event:error
+ * @fires GeolocationError#event:error
  * @api
  */
 class Geolocation extends BaseObject {
