diff --git a/src/ol/webgl.js b/src/ol/webgl.js
index c58714339..471774596 100644
--- a/src/ol/webgl.js
+++ b/src/ol/webgl.js
@@ -53,24 +53,28 @@ export const DYNAMIC_DRAW = 0x88e8;
 /**
  * @const
  * @type {number}
+ * @api
  */
 export const UNSIGNED_BYTE = 0x1401;
 
 /**
  * @const
  * @type {number}
+ * @api
  */
 export const UNSIGNED_SHORT = 0x1403;
 
 /**
  * @const
  * @type {number}
+ * @api
  */
 export const UNSIGNED_INT = 0x1405;
 
 /**
  * @const
  * @type {number}
+ * @api
  */
 export const FLOAT = 0x1406;
 
@@ -80,6 +84,7 @@ export const FLOAT = 0x1406;
 /**
  * @const
  * @type {Array<string>}
+ * @api
  */
 const CONTEXT_IDS = ['experimental-webgl', 'webgl', 'webkit-3d', 'moz-webgl'];
 
@@ -87,6 +92,7 @@ const CONTEXT_IDS = ['experimental-webgl', 'webgl', 'webkit-3d', 'moz-webgl'];
  * @param {HTMLCanvasElement} canvas Canvas.
  * @param {Object=} opt_attributes Attributes.
  * @return {WebGLRenderingContext} WebGL rendering context.
+ * @api
  */
 export function getContext(canvas, opt_attributes) {
   const ii = CONTEXT_IDS.length;
@@ -110,6 +116,7 @@ let supportedExtensions;
 
 /**
  * @return {Array<string>} List of supported WebGL extensions.
+ * @api
  */
 export function getSupportedExtensions() {
   if (!supportedExtensions) {
