diff --git a/src/ol/source/Vector.js b/src/ol/source/Vector.js
index 0d0905e5c..bcf371907 100644
--- a/src/ol/source/Vector.js
+++ b/src/ol/source/Vector.js
@@ -35,13 +35,13 @@ import RBush from '../structs/RBush.js';
  * @classdesc
  * Events emitted by {@link module:ol/source/Vector} instances are instances of this
  * type.
- * @template {import("../geom/Geometry.js").default} Geometry
+ * @template {import("../geom/Geometry.js").default} GeomType
  */
 export class VectorSourceEvent extends Event {
 
   /**
    * @param {string} type Type.
-   * @param {import("../Feature.js").default<Geometry>=} opt_feature Feature.
+   * @param {import("../Feature.js").default<GeomType>=} opt_feature Feature.
    */
   constructor(type, opt_feature) {
 
@@ -49,7 +49,7 @@ export class VectorSourceEvent extends Event {
 
     /**
      * The feature being added or removed.
-     * @type {import("../Feature.js").default<Geometry>|undefined}
+     * @type {import("../Feature.js").default<GeomType>|undefined}
      * @api
      */
     this.feature = opt_feature;
@@ -157,7 +157,7 @@ export class VectorSourceEvent extends Event {
  *
  * @fires VectorSourceEvent
  * @api
- * @template {import("../geom/Geometry.js").default} Geometry
+ * @template {import("../geom/Geometry.js").default} GeomType
  */
 class VectorSource extends Source {
   /**
@@ -217,7 +217,7 @@ class VectorSource extends Source {
 
     /**
      * @private
-     * @type {RBush<import("../Feature.js").default<Geometry>>}
+     * @type {RBush<import("../Feature.js").default<GeomType>>}
      */
     this.featuresRtree_ = useSpatialIndex ? new RBush() : null;
 
@@ -229,21 +229,21 @@ class VectorSource extends Source {
 
     /**
      * @private
-     * @type {!Object<string, import("../Feature.js").default<Geometry>>}
+     * @type {!Object<string, import("../Feature.js").default<GeomType>>}
      */
     this.nullGeometryFeatures_ = {};
 
     /**
      * A lookup of features by id (the return from feature.getId()).
      * @private
-     * @type {!Object<string, import("../Feature.js").default<Geometry>>}
+     * @type {!Object<string, import("../Feature.js").default<GeomType>>}
      */
     this.idIndex_ = {};
 
     /**
      * A lookup of features by uid (using getUid(feature)).
      * @private
-     * @type {!Object<string, import("../Feature.js").default<Geometry>>}
+     * @type {!Object<string, import("../Feature.js").default<GeomType>>}
      */
     this.uidIndex_ = {};
 
@@ -255,7 +255,7 @@ class VectorSource extends Source {
 
     /**
      * @private
-     * @type {Collection<import("../Feature.js").default<Geometry>>}
+     * @type {Collection<import("../Feature.js").default<GeomType>>}
      */
     this.featuresCollection_ = null;
 
@@ -287,7 +287,7 @@ class VectorSource extends Source {
    * Note: this also applies if an {@link module:ol/Collection} is used for features,
    * meaning that if a feature with a duplicate id is added in the collection, it will
    * be removed from it right away.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature to add.
+   * @param {import("../Feature.js").default<GeomType>} feature Feature to add.
    * @api
    */
   addFeature(feature) {
@@ -298,7 +298,7 @@ class VectorSource extends Source {
 
   /**
    * Add a feature without firing a `change` event.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature.
+   * @param {import("../Feature.js").default<GeomType>} feature Feature.
    * @protected
    */
   addFeatureInternal(feature) {
@@ -330,7 +330,7 @@ class VectorSource extends Source {
 
   /**
    * @param {string} featureKey Unique identifier for the feature.
-   * @param {import("../Feature.js").default<Geometry>} feature The feature.
+   * @param {import("../Feature.js").default<GeomType>} feature The feature.
    * @private
    */
   setupChangeEvents_(featureKey, feature) {
@@ -345,7 +345,7 @@ class VectorSource extends Source {
 
   /**
    * @param {string} featureKey Unique identifier for the feature.
-   * @param {import("../Feature.js").default<Geometry>} feature The feature.
+   * @param {import("../Feature.js").default<GeomType>} feature The feature.
    * @return {boolean} The feature is "valid", in the sense that it is also a
    *     candidate for insertion into the Rtree.
    * @private
@@ -371,7 +371,7 @@ class VectorSource extends Source {
 
   /**
    * Add a batch of features to the source.
-   * @param {Array<import("../Feature.js").default<Geometry>>} features Features to add.
+   * @param {Array<import("../Feature.js").default<GeomType>>} features Features to add.
    * @api
    */
   addFeatures(features) {
@@ -382,7 +382,7 @@ class VectorSource extends Source {
 
   /**
    * Add features without firing a `change` event.
-   * @param {Array<import("../Feature.js").default<Geometry>>} features Features.
+   * @param {Array<import("../Feature.js").default<GeomType>>} features Features.
    * @protected
    */
   addFeaturesInternal(features) {
@@ -423,14 +423,14 @@ class VectorSource extends Source {
 
 
   /**
-   * @param {!Collection<import("../Feature.js").default<Geometry>>} collection Collection.
+   * @param {!Collection<import("../Feature.js").default<GeomType>>} collection Collection.
    * @private
    */
   bindFeaturesCollection_(collection) {
     let modifyingCollection = false;
     this.addEventListener(VectorEventType.ADDFEATURE,
       /**
-       * @param {VectorSourceEvent<Geometry>} evt The vector source event
+       * @param {VectorSourceEvent<GeomType>} evt The vector source event
        */
       function(evt) {
         if (!modifyingCollection) {
@@ -441,7 +441,7 @@ class VectorSource extends Source {
       });
     this.addEventListener(VectorEventType.REMOVEFEATURE,
       /**
-       * @param {VectorSourceEvent<Geometry>} evt The vector source event
+       * @param {VectorSourceEvent<GeomType>} evt The vector source event
        */
       function(evt) {
         if (!modifyingCollection) {
@@ -457,7 +457,7 @@ class VectorSource extends Source {
       function(evt) {
         if (!modifyingCollection) {
           modifyingCollection = true;
-          this.addFeature(/** @type {import("../Feature.js").default<Geometry>} */ (evt.element));
+          this.addFeature(/** @type {import("../Feature.js").default<GeomType>} */ (evt.element));
           modifyingCollection = false;
         }
       }.bind(this));
@@ -468,7 +468,7 @@ class VectorSource extends Source {
       function(evt) {
         if (!modifyingCollection) {
           modifyingCollection = true;
-          this.removeFeature(/** @type {import("../Feature.js").default<Geometry>} */ (evt.element));
+          this.removeFeature(/** @type {import("../Feature.js").default<GeomType>} */ (evt.element));
           modifyingCollection = false;
         }
       }.bind(this));
@@ -521,7 +521,7 @@ class VectorSource extends Source {
    * stop and the function will return the same value.
    * Note: this function only iterate through the feature that have a defined geometry.
    *
-   * @param {function(import("../Feature.js").default<Geometry>): T} callback Called with each feature
+   * @param {function(import("../Feature.js").default<GeomType>): T} callback Called with each feature
    *     on the source.  Return a truthy value to stop iteration.
    * @return {T|undefined} The return value from the last call to the callback.
    * @template T
@@ -543,7 +543,7 @@ class VectorSource extends Source {
    * value.
    *
    * @param {import("../coordinate.js").Coordinate} coordinate Coordinate.
-   * @param {function(import("../Feature.js").default<Geometry>): T} callback Called with each feature
+   * @param {function(import("../Feature.js").default<GeomType>): T} callback Called with each feature
    *     whose goemetry contains the provided coordinate.
    * @return {T|undefined} The return value from the last call to the callback.
    * @template T
@@ -574,7 +574,7 @@ class VectorSource extends Source {
    * features, equivalent to {@link module:ol/source/Vector~VectorSource#forEachFeature #forEachFeature()}.
    *
    * @param {import("../extent.js").Extent} extent Extent.
-   * @param {function(import("../Feature.js").default<Geometry>): T} callback Called with each feature
+   * @param {function(import("../Feature.js").default<GeomType>): T} callback Called with each feature
    *     whose bounding box intersects the provided extent.
    * @return {T|undefined} The return value from the last call to the callback.
    * @template T
@@ -598,7 +598,7 @@ class VectorSource extends Source {
    * {@link module:ol/source/Vector~VectorSource#forEachFeatureInExtent #forEachFeatureInExtent()} method instead.
    *
    * @param {import("../extent.js").Extent} extent Extent.
-   * @param {function(import("../Feature.js").default<Geometry>): T} callback Called with each feature
+   * @param {function(import("../Feature.js").default<GeomType>): T} callback Called with each feature
    *     whose geometry intersects the provided extent.
    * @return {T|undefined} The return value from the last call to the callback.
    * @template T
@@ -607,7 +607,7 @@ class VectorSource extends Source {
   forEachFeatureIntersectingExtent(extent, callback) {
     return this.forEachFeatureInExtent(extent,
       /**
-       * @param {import("../Feature.js").default<Geometry>} feature Feature.
+       * @param {import("../Feature.js").default<GeomType>} feature Feature.
        * @return {T|undefined} The return value from the last call to the callback.
        */
       function(feature) {
@@ -626,7 +626,7 @@ class VectorSource extends Source {
    * Get the features collection associated with this source. Will be `null`
    * unless the source was configured with `useSpatialIndex` set to `false`, or
    * with an {@link module:ol/Collection} as `features`.
-   * @return {Collection<import("../Feature.js").default<Geometry>>} The collection of features.
+   * @return {Collection<import("../Feature.js").default<GeomType>>} The collection of features.
    * @api
    */
   getFeaturesCollection() {
@@ -636,7 +636,7 @@ class VectorSource extends Source {
 
   /**
    * Get all features on the source in random order.
-   * @return {Array<import("../Feature.js").default<Geometry>>} Features.
+   * @return {Array<import("../Feature.js").default<GeomType>>} Features.
    * @api
    */
   getFeatures() {
@@ -650,7 +650,7 @@ class VectorSource extends Source {
       }
     }
     return (
-      /** @type {Array<import("../Feature.js").default<Geometry>>} */ (features)
+      /** @type {Array<import("../Feature.js").default<GeomType>>} */ (features)
     );
   }
 
@@ -658,7 +658,7 @@ class VectorSource extends Source {
   /**
    * Get all features whose geometry intersects the provided coordinate.
    * @param {import("../coordinate.js").Coordinate} coordinate Coordinate.
-   * @return {Array<import("../Feature.js").default<Geometry>>} Features.
+   * @return {Array<import("../Feature.js").default<GeomType>>} Features.
    * @api
    */
   getFeaturesAtCoordinate(coordinate) {
@@ -679,7 +679,7 @@ class VectorSource extends Source {
    * features.
    *
    * @param {import("../extent.js").Extent} extent Extent.
-   * @return {Array<import("../Feature.js").default<Geometry>>} Features.
+   * @return {Array<import("../Feature.js").default<GeomType>>} Features.
    * @api
    */
   getFeaturesInExtent(extent) {
@@ -699,10 +699,10 @@ class VectorSource extends Source {
    * This method is not available when the source is configured with
    * `useSpatialIndex` set to `false`.
    * @param {import("../coordinate.js").Coordinate} coordinate Coordinate.
-   * @param {function(import("../Feature.js").default<Geometry>):boolean=} opt_filter Feature filter function.
+   * @param {function(import("../Feature.js").default<GeomType>):boolean=} opt_filter Feature filter function.
    *     The filter function will receive one argument, the {@link module:ol/Feature feature}
    *     and it should return a boolean value. By default, no filtering is made.
-   * @return {import("../Feature.js").default<Geometry>} Closest feature.
+   * @return {import("../Feature.js").default<GeomType>} Closest feature.
    * @api
    */
   getClosestFeatureToCoordinate(coordinate, opt_filter) {
@@ -722,7 +722,7 @@ class VectorSource extends Source {
     const filter = opt_filter ? opt_filter : TRUE;
     this.featuresRtree_.forEachInExtent(extent,
       /**
-       * @param {import("../Feature.js").default<Geometry>} feature Feature.
+       * @param {import("../Feature.js").default<GeomType>} feature Feature.
        */
       function(feature) {
         if (filter(feature)) {
@@ -769,7 +769,7 @@ class VectorSource extends Source {
    * `source.getFeatureById(2)` will return a feature with id `'2'` or `2`.
    *
    * @param {string|number} id Feature identifier.
-   * @return {import("../Feature.js").default<Geometry>} The feature (or `null` if not found).
+   * @return {import("../Feature.js").default<GeomType>} The feature (or `null` if not found).
    * @api
    */
   getFeatureById(id) {
@@ -782,7 +782,7 @@ class VectorSource extends Source {
    * Get a feature by its internal unique identifier (using `getUid`).
    *
    * @param {string} uid Feature identifier.
-   * @return {import("../Feature.js").default<Geometry>} The feature (or `null` if not found).
+   * @return {import("../Feature.js").default<GeomType>} The feature (or `null` if not found).
    */
   getFeatureByUid(uid) {
     const feature = this.uidIndex_[uid];
@@ -825,7 +825,7 @@ class VectorSource extends Source {
    * @private
    */
   handleFeatureChange_(event) {
-    const feature = /** @type {import("../Feature.js").default<Geometry>} */ (event.target);
+    const feature = /** @type {import("../Feature.js").default<GeomType>} */ (event.target);
     const featureKey = getUid(feature);
     const geometry = feature.getGeometry();
     if (!geometry) {
@@ -866,7 +866,7 @@ class VectorSource extends Source {
 
   /**
    * Returns true if the feature is contained within the source.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature.
+   * @param {import("../Feature.js").default<GeomType>} feature Feature.
    * @return {boolean} Has feature.
    * @api
    */
@@ -948,7 +948,7 @@ class VectorSource extends Source {
    * Remove a single feature from the source.  If you want to remove all features
    * at once, use the {@link module:ol/source/Vector~VectorSource#clear #clear()} method
    * instead.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature to remove.
+   * @param {import("../Feature.js").default<GeomType>} feature Feature to remove.
    * @api
    */
   removeFeature(feature) {
@@ -967,7 +967,7 @@ class VectorSource extends Source {
 
   /**
    * Remove feature without firing a `change` event.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature.
+   * @param {import("../Feature.js").default<GeomType>} feature Feature.
    * @protected
    */
   removeFeatureInternal(feature) {
@@ -987,7 +987,7 @@ class VectorSource extends Source {
   /**
    * Remove a feature from the id index.  Called internally when the feature id
    * may have changed.
-   * @param {import("../Feature.js").default<Geometry>} feature The feature.
+   * @param {import("../Feature.js").default<GeomType>} feature The feature.
    * @return {boolean} Removed the feature from the index.
    * @private
    */
