diff --git a/src/ol/renderer/canvas/ImageLayer.js b/src/ol/renderer/canvas/ImageLayer.js
index 72e6933b9..667f11a2b 100644
--- a/src/ol/renderer/canvas/ImageLayer.js
+++ b/src/ol/renderer/canvas/ImageLayer.js
@@ -217,6 +217,16 @@ class CanvasImageLayerRenderer extends CanvasLayerRenderer {
 
     return this.container;
   }
+
+  /**
+   * @inheritDoc
+   */
+  forEachFeatureAtCoordinate(coordinate, frameState, hitTolerance, callback, declutteredFeatures) {}
+
+  /**
+   * @inheritDoc
+   */
+  handleFontsChanged() {}
 }
 
 export default CanvasImageLayerRenderer;
