diff --git a/src/ol/Collection.js b/src/ol/Collection.js
index bf5f0f918..bcb551a84 100644
--- a/src/ol/Collection.js
+++ b/src/ol/Collection.js
@@ -20,6 +20,9 @@ const Property = {
  * @classdesc
  * Events emitted by {@link module:ol/Collection~Collection} instances are instances of this
  * type.
+ *
+ * @api
+ * @template T
  */
 export class CollectionEvent extends Event {
 
@@ -33,7 +36,7 @@ export class CollectionEvent extends Event {
 
     /**
      * The element that is added to or removed from the collection.
-     * @type {*}
+     * @type {T}
      * @api
      */
     this.element = opt_element;
