diff --git a/src/ol/control/FullScreen.js b/src/ol/control/FullScreen.js
index 05002b01a..fe81bb2a3 100644
--- a/src/ol/control/FullScreen.js
+++ b/src/ol/control/FullScreen.js
@@ -35,9 +35,9 @@ const FullScreenEventType = {
 /**
  * @typedef {Object} Options
  * @property {string} [className='ol-full-screen'] CSS class name.
- * @property {string|Text} [label='\u2922'] Text label to use for the button.
+ * @property {string|Text|HTMLElement} [label='\u2922'] Text label to use for the button.
  * Instead of text, also an element (e.g. a `span` element) can be used.
- * @property {string|Text} [labelActive='\u00d7'] Text label to use for the
+ * @property {string|Text|HTMLElement} [labelActive='\u00d7'] Text label to use for the
  * button when full-screen is active.
  * @property {string} [activeClassName=className + '-true'] CSS class name for the button
  * when full-screen is active.
@@ -64,8 +64,8 @@ const FullScreenEventType = {
  * The [Fullscreen API](http://www.w3.org/TR/fullscreen/) is used to
  * toggle the map in full screen mode.
  *
- * @fires FullScreenEventType#enterfullscreen
- * @fires FullScreenEventType#leavefullscreen
+ * @fires module:ol/events/Event~BaseEvent#event:enterfullscreen
+ * @fires module:ol/events/Event~BaseEvent#event:leavefullscreen
  * @api
  */
 class FullScreen extends Control {
