diff --git a/src/ol/Feature.js b/src/ol/Feature.js
index 7246a0a11..405804149 100644
--- a/src/ol/Feature.js
+++ b/src/ol/Feature.js
@@ -174,7 +174,7 @@ class Feature extends BaseObject {
   /**
    * Get the feature's style. Will return what was provided to the
    * {@link module:ol/Feature~Feature#setStyle} method.
-   * @return {import("./style/Style.js").StyleLike} The feature style.
+   * @return {import("./style/Style.js").StyleLike|null} The feature style.
    * @api
    */
   getStyle() {
@@ -229,7 +229,7 @@ class Feature extends BaseObject {
    * Set the style for the feature.  This can be a single style object, an array
    * of styles, or a function that takes a resolution and returns an array of
    * styles. If it is `null` the feature has no style (a `null` style).
-   * @param {import("./style/Style.js").StyleLike} style Style for this feature.
+   * @param {import("./style/Style.js").StyleLike|null} style Style for this feature.
    * @api
    * @fires module:ol/events/Event~BaseEvent#event:change
    */
diff --git a/src/ol/layer/BaseVector.js b/src/ol/layer/BaseVector.js
index 8c2024ffd..4857072e0 100644
--- a/src/ol/layer/BaseVector.js
+++ b/src/ol/layer/BaseVector.js
@@ -155,7 +155,7 @@ class BaseVectorLayer extends Layer {
   /**
    * Get the style for features.  This returns whatever was passed to the `style`
    * option at construction or to the `setStyle` method.
-   * @return {import("../style/Style.js").StyleLike}
+   * @return {import("../style/Style.js").StyleLike|null|undefined}
    *     Layer style.
    * @api
    */
@@ -203,7 +203,7 @@ class BaseVectorLayer extends Layer {
    * it is `null` the layer has no style (a `null` style), so only features
    * that have their own styles will be rendered in the layer. See
    * {@link module:ol/style} for information on the default style.
-   * @param {import("../style/Style.js").default|Array<import("../style/Style.js").default>|import("../style/Style.js").StyleFunction|null|undefined} style Layer style.
+   * @param {import("../style/Style.js").StyleLike|null|undefined} style Layer style.
    * @api
    */
   setStyle(style) {
