diff --git a/src/ol/renderer/webgl/PointsLayer.js b/src/ol/renderer/webgl/PointsLayer.js
index da3d9dcb1..f77eff190 100644
--- a/src/ol/renderer/webgl/PointsLayer.js
+++ b/src/ol/renderer/webgl/PointsLayer.js
@@ -39,7 +39,7 @@ import {listen, unlistenByKey} from '../../events.js';
  * rebuildBuffers by accessing these objects quicker.
  * @property {import("../../Feature").default} feature Feature
  * @property {Object<string, *>} properties Feature properties
- * @property {import("../../geom").Geometry} geometry Feature geometry
+ * @property {import("../../geom/Geometry").default} geometry Feature geometry
  */
 
 /**
