diff --git a/src/ol/layer/MapboxVector.js b/src/ol/layer/MapboxVector.js
index 370e17bd4..9992362c9 100644
--- a/src/ol/layer/MapboxVector.js
+++ b/src/ol/layer/MapboxVector.js
@@ -139,7 +139,7 @@ class ErrorEvent extends BaseEvent {
  * The Mapbox source type.
  * @enum {string}
  */
-const SourceType = {
+export const SourceType = {
   VECTOR: 'vector',
 };
 
diff --git a/src/ol/source/TileDebug.js b/src/ol/source/TileDebug.js
index f662f158f..668137bbf 100644
--- a/src/ol/source/TileDebug.js
+++ b/src/ol/source/TileDebug.js
@@ -9,7 +9,7 @@ import {createCanvasContext2D} from '../dom.js';
 import {getKeyZXY} from '../tilecoord.js';
 import {toSize} from '../size.js';
 
-class LabeledTile extends Tile {
+export class LabeledTile extends Tile {
   /**
    * @param {import("../tilecoord.js").TileCoord} tileCoord Tile coordinate.
    * @param {import("../size.js").Size} tileSize Tile size.
