diff --git a/src/ol/interaction/Draw.js b/src/ol/interaction/Draw.js
index 5848ab2f3..9c880caee 100644
--- a/src/ol/interaction/Draw.js
+++ b/src/ol/interaction/Draw.js
@@ -109,7 +109,7 @@ import {fromUserCoordinate, getUserProjection} from '../proj.js';
  * and a projection as arguments, and returns a geometry. The optional existing
  * geometry is the geometry that is returned when the function is called without
  * a second argument.
- * @typedef {function(!SketchCoordType, import("../geom/SimpleGeometry.js").default=,
+ * @typedef {function(!SketchCoordType, import("../geom/SimpleGeometry.js").default|undefined,
  *     import("../proj/Projection.js").default):
  *     import("../geom/SimpleGeometry.js").default} GeometryFunction
  */
