diff --git a/src/ol/Geolocation.js b/src/ol/Geolocation.js
index 26f76d229..cb5147591 100644
--- a/src/ol/Geolocation.js
+++ b/src/ol/Geolocation.js
@@ -28,13 +28,22 @@ const Property = {
   TRACKING_OPTIONS: 'trackingOptions',
 };
 
+/**
+ * @typedef {Object} GeolocationPositionError
+ * @property {number} code
+ * @property {string} message
+ * @property {number} PERMISSION_DENIED
+ * @property {number} POSITION_UNAVAILABLE
+ * @property {number} TIMEOUT
+ */
+
 /**
  * @classdesc
  * Events emitted on Geolocation error.
  */
 class GeolocationError extends BaseEvent {
   /**
-   * @param {PositionError} error error object.
+   * @param {GeolocationPositionError} error error object.
    */
   constructor(error) {
     super(EventType.ERROR);
