diff --git a/src/ol/control/FullScreen.js b/src/ol/control/FullScreen.js
index 3c21dcd07..cbbe834b1 100644
--- a/src/ol/control/FullScreen.js
+++ b/src/ol/control/FullScreen.js
@@ -73,8 +73,8 @@ const FullScreenEventType = {
  * The [Fullscreen API](https://www.w3.org/TR/fullscreen/) is used to
  * toggle the map in full screen mode.
  *
- * @fires FullScreenEventType#enterfullscreen
- * @fires FullScreenEventType#leavefullscreen
+ * @fires module:ol/events/Event~BaseEvent#event:enterfullscreen
+ * @fires module:ol/events/Event~BaseEvent#event:leavefullscreen
  * @api
  */
 class FullScreen extends Control {
