diff --git a/src/ol/Feature.js b/src/ol/Feature.js
index 834e618fa..6b168fc9e 100644
--- a/src/ol/Feature.js
+++ b/src/ol/Feature.js
@@ -65,11 +65,11 @@ import {listen, unlistenByKey} from './events.js';
  * ```
  *
  * @api
- * @template {import("./geom/Geometry.js").default} Geometry
+ * @template {import("./geom/Geometry.js").default} G
  */
 class Feature extends BaseObject {
   /**
-   * @param {Geometry|Object<string, *>} [opt_geometryOrProperties]
+   * @param {G|Object<string, *>} [opt_geometryOrProperties]
    *     You may pass a Geometry object directly, or an object literal containing
    *     properties. If you pass an object literal, you may include a Geometry
    *     associated with a `geometry` key.
@@ -131,7 +131,7 @@ class Feature extends BaseObject {
           /** @type {?} */ (opt_geometryOrProperties).getSimplifiedGeometry
         ) === 'function'
       ) {
-        const geometry = /** @type {Geometry} */ (opt_geometryOrProperties);
+        const geometry = /** @type {G} */ (opt_geometryOrProperties);
         this.setGeometry(geometry);
       } else {
         /** @type {Object<string, *>} */
@@ -167,12 +167,12 @@ class Feature extends BaseObject {
    * Get the feature's default geometry.  A feature may have any number of named
    * geometries.  The "default" geometry (the one that is rendered by default) is
    * set when calling {@link module:ol/Feature~Feature#setGeometry}.
-   * @return {Geometry|undefined} The default geometry for the feature.
+   * @return {G|undefined} The default geometry for the feature.
    * @api
    * @observable
    */
   getGeometry() {
-    return /** @type {Geometry|undefined} */ (this.get(this.geometryName_));
+    return /** @type {G|undefined} */ (this.get(this.geometryName_));
   }
 
   /**
@@ -247,7 +247,7 @@ class Feature extends BaseObject {
   /**
    * Set the default geometry for the feature.  This will update the property
    * with the name returned by {@link module:ol/Feature~Feature#getGeometryName}.
-   * @param {Geometry|undefined} geometry The new geometry.
+   * @param {G|undefined} geometry The new geometry.
    * @api
    * @observable
    */
