diff --git a/src/ol/layer/BaseImage.js b/src/ol/layer/BaseImage.js
index e6ddfcc0c..880b8c38e 100644
--- a/src/ol/layer/BaseImage.js
+++ b/src/ol/layer/BaseImage.js
@@ -41,7 +41,7 @@ import Layer from './Layer.js';
  *
  * @template {import("../source/Image.js").default} ImageSourceType
  * @template {import("../renderer/Layer.js").default} RendererType
- * @extends {Layer<ImageSourceType, RendererType>}
+ * @augments Layer<ImageSourceType,RendererType>
  * @api
  */
 class BaseImageLayer extends Layer {
diff --git a/src/ol/layer/BaseTile.js b/src/ol/layer/BaseTile.js
index d0d68cafd..2e7d1ac34 100644
--- a/src/ol/layer/BaseTile.js
+++ b/src/ol/layer/BaseTile.js
@@ -55,7 +55,7 @@ import TileProperty from './TileProperty.js';
  *
  * @template {import("../source/Tile.js").default} TileSourceType
  * @template {import("../renderer/Layer.js").default} RendererType
- * @extends {Layer<TileSourceType, RendererType>}
+ * @augments Layer<TileSourceType,RendererType>
  * @api
  */
 class BaseTileLayer extends Layer {
diff --git a/src/ol/layer/BaseVector.js b/src/ol/layer/BaseVector.js
index 375bde86e..346206fed 100644
--- a/src/ol/layer/BaseVector.js
+++ b/src/ol/layer/BaseVector.js
@@ -81,7 +81,7 @@ const Property = {
  *
  * @template {import("../source/Vector.js").default|import("../source/VectorTile.js").default} VectorSourceType
  * @template {import("../renderer/canvas/VectorLayer.js").default|import("../renderer/canvas/VectorTileLayer.js").default|import("../renderer/canvas/VectorImageLayer.js").default|import("../renderer/webgl/PointsLayer.js").default} RendererType
- * @extends {Layer<VectorSourceType, RendererType>}
+ * @augments Layer<VectorSourceType,RendererType>
  * @api
  */
 class BaseVectorLayer extends Layer {
diff --git a/src/ol/layer/Graticule.js b/src/ol/layer/Graticule.js
index 57b2cd023..a190ca5ab 100644
--- a/src/ol/layer/Graticule.js
+++ b/src/ol/layer/Graticule.js
@@ -178,7 +178,7 @@ const INTERVALS = [
  * Note that the view projection must define both extent and worldExtent.
  *
  * @fires import("../render/Event.js").RenderEvent
- * @extends {VectorLayer<import("../source/Vector.js").default>}
+ * @augments VectorLayer<import("../source/Vector.js").default>
  * @api
  */
 class Graticule extends VectorLayer {
diff --git a/src/ol/layer/Heatmap.js b/src/ol/layer/Heatmap.js
index 03a6f125e..c79071c80 100644
--- a/src/ol/layer/Heatmap.js
+++ b/src/ol/layer/Heatmap.js
@@ -60,7 +60,7 @@ const DEFAULT_GRADIENT = ['#00f', '#0ff', '#0f0', '#ff0', '#f00'];
  * options means that `title` is observable, and has get/set accessors.
  *
  * @fires import("../render/Event.js").RenderEvent
- * @extends {BaseVector<import("../source/Vector.js").default, WebGLPointsLayerRenderer>}
+ * @augments BaseVector<import("../source/Vector.js").default,import("../renderer/webgl/PointsLayer.js").default>
  * @api
  */
 class Heatmap extends BaseVector {
diff --git a/src/ol/layer/Image.js b/src/ol/layer/Image.js
index 3f7bf2499..83e0fc3af 100644
--- a/src/ol/layer/Image.js
+++ b/src/ol/layer/Image.js
@@ -13,7 +13,7 @@ import CanvasImageLayerRenderer from '../renderer/canvas/ImageLayer.js';
  * options means that `title` is observable, and has get/set accessors.
  *
  * @template {import("../source/Image.js").default} ImageSourceType
- * @extends {BaseImageLayer<ImageSourceType, CanvasImageLayerRenderer>}
+ * @augments BaseImageLayer<ImageSourceType,import("../renderer/canvas/ImageLayer.js").default>
  * @api
  */
 class ImageLayer extends BaseImageLayer {
diff --git a/src/ol/layer/MapboxVector.js b/src/ol/layer/MapboxVector.js
index 5c1d26d4d..67e0e33d2 100644
--- a/src/ol/layer/MapboxVector.js
+++ b/src/ol/layer/MapboxVector.js
@@ -134,7 +134,7 @@ class ErrorEvent extends BaseEvent {
  * [ol-mapbox-style](https://github.com/openlayers/ol-mapbox-style) library to be loaded as well.
  *
  * @param {Options} options Options.
- * @extends {VectorTileLayer}
+ * @augments VectorTileLayer
  * @fires module:ol/events/Event~BaseEvent#event:error
  * @api
  */
diff --git a/src/ol/layer/Tile.js b/src/ol/layer/Tile.js
index da459d6a1..502d71853 100644
--- a/src/ol/layer/Tile.js
+++ b/src/ol/layer/Tile.js
@@ -13,7 +13,7 @@ import CanvasTileLayerRenderer from '../renderer/canvas/TileLayer.js';
  * options means that `title` is observable, and has get/set accessors.
  *
  * @template {import("../source/Tile.js").default} TileSourceType
- * @extends BaseTileLayer<TileSourceType, CanvasTileLayerRenderer>
+ * @augments BaseTileLayer<TileSourceType,import("../renderer/canvas/TileLayer.js").default>
  * @api
  */
 class TileLayer extends BaseTileLayer {
diff --git a/src/ol/layer/Vector.js b/src/ol/layer/Vector.js
index 557362074..13084d52b 100644
--- a/src/ol/layer/Vector.js
+++ b/src/ol/layer/Vector.js
@@ -16,7 +16,7 @@ import CanvasVectorLayerRenderer from '../renderer/canvas/VectorLayer.js';
  * options means that `title` is observable, and has get/set accessors.
  *
  * @template {import("../source/Vector.js").default} VectorSourceType
- * @extends {BaseVectorLayer<VectorSourceType, CanvasVectorLayerRenderer>}
+ * @augments BaseVectorLayer<VectorSourceType,import("../renderer/canvas/VectorLayer.js").default>
  * @api
  */
 class VectorLayer extends BaseVectorLayer {
diff --git a/src/ol/layer/VectorImage.js b/src/ol/layer/VectorImage.js
index c69564b66..466da655a 100644
--- a/src/ol/layer/VectorImage.js
+++ b/src/ol/layer/VectorImage.js
@@ -58,7 +58,7 @@ import CanvasVectorImageLayerRenderer from '../renderer/canvas/VectorImageLayer.
  * options means that `title` is observable, and has get/set accessors.
  *
  * @template {import("../source/Vector.js").default} VectorSourceType
- * @extends {BaseVectorLayer<VectorSourceType, CanvasVectorImageLayerRenderer>}
+ * @augments BaseVectorLayer<VectorSourceType,import("../renderer/canvas/VectorImageLayer.js").default>
  * @api
  */
 class VectorImageLayer extends BaseVectorLayer {
diff --git a/src/ol/layer/VectorTile.js b/src/ol/layer/VectorTile.js
index f98efc039..42855c202 100644
--- a/src/ol/layer/VectorTile.js
+++ b/src/ol/layer/VectorTile.js
@@ -94,7 +94,7 @@ import {assert} from '../asserts.js';
  * options means that `title` is observable, and has get/set accessors.
  *
  * @param {Options} [options] Options.
- * @extends {BaseVectorLayer<import("../source/VectorTile.js").default, CanvasVectorTileLayerRenderer>}
+ * @augments BaseVectorLayer<import("../source/VectorTile.js").default,import("../renderer/canvas/VectorTileLayer.js").default>
  * @api
  */
 class VectorTileLayer extends BaseVectorLayer {
diff --git a/src/ol/layer/WebGLPoints.js b/src/ol/layer/WebGLPoints.js
index 721d761bc..6db1ea007 100644
--- a/src/ol/layer/WebGLPoints.js
+++ b/src/ol/layer/WebGLPoints.js
@@ -68,7 +68,7 @@ import {parseLiteralStyle} from '../webgl/ShaderBuilder.js';
  * options means that `title` is observable, and has get/set accessors.
  *
  * @template {import("../source/Vector.js").default<import("../geom/Point.js").default>} VectorSourceType
- * @extends {Layer<VectorSourceType, WebGLPointsLayerRenderer>}
+ * @augments Layer<VectorSourceType,import("../renderer/webgl/PointsLayer.js").default>
  * @fires import("../render/Event.js").RenderEvent
  */
 class WebGLPointsLayer extends Layer {
diff --git a/src/ol/layer/WebGLTile.js b/src/ol/layer/WebGLTile.js
index 2cf948e0e..efb30c619 100644
--- a/src/ol/layer/WebGLTile.js
+++ b/src/ol/layer/WebGLTile.js
@@ -298,7 +298,7 @@ function parseStyle(style, bandCount) {
  * property on the layer object; for example, setting `title: 'My Title'` in the
  * options means that `title` is observable, and has get/set accessors.
  *
- * @extends BaseTileLayer<SourceType, WebGLTileLayerRenderer>
+ * @augments BaseTileLayer<SourceType,import("../renderer/webgl/TileLayer.js").default>
  * @fires import("../render/Event.js").RenderEvent
  * @api
  */
