diff --git a/src/ol/obj.js b/src/ol/obj.js
index 428fbf09a..55f655f89 100644
--- a/src/ol/obj.js
+++ b/src/ol/obj.js
@@ -49,7 +49,7 @@ export function clear(object) {
  *
  * @param {!Object<K,V>} object The object from which to get the values.
  * @return {!Array<V>} The property values.
- * @template K,V
+ * @template {string|number|symbol} K, V
  */
 export const getValues =
   typeof Object.values === 'function'
