diff --git a/src/ol/renderer/canvas/common.js b/src/ol/renderer/canvas/common.js
index 81d185bef..2eb4827c7 100644
--- a/src/ol/renderer/canvas/common.js
+++ b/src/ol/renderer/canvas/common.js
@@ -2,9 +2,15 @@
  * @module ol/renderer/canvas/common
  */
 
+/**
+ * @typedef ImageSmoothingContext
+ * @property {boolean} imageSmoothingEnabled
+ * @property {boolean} msImageSmoothingEnabled
+ */
+
 /**
  * Context options to disable image smoothing.
- * @type {Object}
+ * @type {ImageSmoothingContext}
  */
 export const IMAGE_SMOOTHING_DISABLED = {
   imageSmoothingEnabled: false,
@@ -13,7 +19,7 @@ export const IMAGE_SMOOTHING_DISABLED = {
 
 /**
  * Context options to enable image smoothing.
- * @type {Object}
+ * @type {ImageSmoothingContext}
  */
 export const IMAGE_SMOOTHING_ENABLED = {
   imageSmoothingEnabled: true,
