diff --git a/src/ol/render/canvas.js b/src/ol/render/canvas.js
index 01025e5d7..e2dccc86d 100644
--- a/src/ol/render/canvas.js
+++ b/src/ol/render/canvas.js
@@ -272,8 +272,9 @@ export const registerFont = (function () {
 })();
 
 /**
+ * @function
  * @param {string} font Font to use for measuring.
- * @return {import("../size.js").Size} Measurement.
+ * @return {number} Measurement.
  */
 export const measureTextHeight = (function () {
   /**
diff --git a/src/ol/render/canvas/BuilderGroup.js b/src/ol/render/canvas/BuilderGroup.js
index b0c121cfd..c6e1f1573 100644
--- a/src/ol/render/canvas/BuilderGroup.js
+++ b/src/ol/render/canvas/BuilderGroup.js
@@ -60,7 +60,7 @@ class BuilderGroup {
   }
 
   /**
-   * @return {!Object<string, !Object<import("../canvas.js").BuilderType, import("./Builder.js").SerializableInstructions>>} The serializable instructions
+   * @return {!Object<string, !Object<import("../canvas.js").BuilderType, import("../canvas").SerializableInstructions>>} The serializable instructions
    */
   finish() {
     const builderInstructions = {};
