diff --git a/src/ol/render/canvas.js b/src/ol/render/canvas.js
index 6202e5f92..8471cd2c5 100644
--- a/src/ol/render/canvas.js
+++ b/src/ol/render/canvas.js
@@ -281,8 +281,9 @@ export const registerFont = (function () {
 })();
 
 /**
+ * @function
  * @param {string} font Font to use for measuring.
- * @return {import("../size.js").Size} Measurement.
+ * @return {number} Measurement.
  */
 export const measureTextHeight = (function () {
   /**
diff --git a/src/ol/render/canvas/BuilderGroup.js b/src/ol/render/canvas/BuilderGroup.js
index 49aa6c44c..4cd5e489a 100644
--- a/src/ol/render/canvas/BuilderGroup.js
+++ b/src/ol/render/canvas/BuilderGroup.js
@@ -60,7 +60,7 @@ class BuilderGroup {
   }
 
   /**
-   * @return {!Object<string, !Object<import("./BuilderType").default, import("./Builder.js").SerializableInstructions>>} The serializable instructions
+   * @return {!Object<string, !Object<import("./BuilderType").default, import("../canvas.js").SerializableInstructions>>} The serializable instructions
    */
   finish() {
     const builderInstructions = {};
