diff --git a/src/ol/source/Vector.js b/src/ol/source/Vector.js
index 1defcf3e1..ff81bd947 100644
--- a/src/ol/source/Vector.js
+++ b/src/ol/source/Vector.js
@@ -34,27 +34,27 @@ import {xhr} from '../featureloader.js';
  * @classdesc
  * Events emitted by {@link module:ol/source/Vector} instances are instances of this
  * type.
- * @template {import("../geom/Geometry.js").default} Geometry
+ * @template {import("../geom/Geometry.js").default} G
  */
 export class VectorSourceEvent extends Event {
   /**
    * @param {string} type Type.
-   * @param {import("../Feature.js").default<Geometry>} [opt_feature] Feature.
-   * @param {Array<import("../Feature.js").default<Geometry>>} [opt_features] Features.
+   * @param {import("../Feature.js").default<G>} [opt_feature] Feature.
+   * @param {Array<import("../Feature.js").default<G>>} [opt_features] Features.
    */
   constructor(type, opt_feature, opt_features) {
     super(type);
 
     /**
      * The added or removed feature for the `ADDFEATURE` and `REMOVEFEATURE` events, `undefined` otherwise.
-     * @type {import("../Feature.js").default<Geometry>|undefined}
+     * @type {import("../Feature.js").default<G>|undefined}
      * @api
      */
     this.feature = opt_feature;
 
     /**
      * The loaded features for the `FEATURESLOADED` event, `undefined` otherwise.
-     * @type {Array<import("../Feature.js").default<Geometry>>|undefined}
+     * @type {Array<import("../Feature.js").default<G>>|undefined}
      * @api
      */
     this.features = opt_features;
@@ -170,7 +170,7 @@ export class VectorSourceEvent extends Event {
  *
  * @fires VectorSourceEvent
  * @api
- * @template {import("../geom/Geometry.js").default} Geometry
+ * @template {import("../geom/Geometry.js").default} G
  */
 class VectorSource extends Source {
   /**
@@ -248,7 +248,7 @@ class VectorSource extends Source {
 
     /**
      * @private
-     * @type {RBush<import("../Feature.js").default<Geometry>>}
+     * @type {RBush<import("../Feature.js").default<G>>}
      */
     this.featuresRtree_ = useSpatialIndex ? new RBush() : null;
 
@@ -266,21 +266,21 @@ class VectorSource extends Source {
 
     /**
      * @private
-     * @type {!Object<string, import("../Feature.js").default<Geometry>>}
+     * @type {!Object<string, import("../Feature.js").default<G>>}
      */
     this.nullGeometryFeatures_ = {};
 
     /**
      * A lookup of features by id (the return from feature.getId()).
      * @private
-     * @type {!Object<string, import("../Feature.js").default<Geometry>>}
+     * @type {!Object<string, import("../Feature.js").default<G>>}
      */
     this.idIndex_ = {};
 
     /**
      * A lookup of features by uid (using getUid(feature)).
      * @private
-     * @type {!Object<string, import("../Feature.js").default<Geometry>>}
+     * @type {!Object<string, import("../Feature.js").default<G>>}
      */
     this.uidIndex_ = {};
 
@@ -292,7 +292,7 @@ class VectorSource extends Source {
 
     /**
      * @private
-     * @type {Collection<import("../Feature.js").default<Geometry>>}
+     * @type {Collection<import("../Feature.js").default<G>>}
      */
     this.featuresCollection_ = null;
 
@@ -323,7 +323,7 @@ class VectorSource extends Source {
    * Note: this also applies if an {@link module:ol/Collection} is used for features,
    * meaning that if a feature with a duplicate id is added in the collection, it will
    * be removed from it right away.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature to add.
+   * @param {import("../Feature.js").default<G>} feature Feature to add.
    * @api
    */
   addFeature(feature) {
@@ -333,7 +333,7 @@ class VectorSource extends Source {
 
   /**
    * Add a feature without firing a `change` event.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature.
+   * @param {import("../Feature.js").default<G>} feature Feature.
    * @protected
    */
   addFeatureInternal(feature) {
@@ -365,7 +365,7 @@ class VectorSource extends Source {
 
   /**
    * @param {string} featureKey Unique identifier for the feature.
-   * @param {import("../Feature.js").default<Geometry>} feature The feature.
+   * @param {import("../Feature.js").default<G>} feature The feature.
    * @private
    */
   setupChangeEvents_(featureKey, feature) {
@@ -382,7 +382,7 @@ class VectorSource extends Source {
 
   /**
    * @param {string} featureKey Unique identifier for the feature.
-   * @param {import("../Feature.js").default<Geometry>} feature The feature.
+   * @param {import("../Feature.js").default<G>} feature The feature.
    * @return {boolean} The feature is "valid", in the sense that it is also a
    *     candidate for insertion into the Rtree.
    * @private
@@ -406,7 +406,7 @@ class VectorSource extends Source {
 
   /**
    * Add a batch of features to the source.
-   * @param {Array<import("../Feature.js").default<Geometry>>} features Features to add.
+   * @param {Array<import("../Feature.js").default<G>>} features Features to add.
    * @api
    */
   addFeatures(features) {
@@ -416,7 +416,7 @@ class VectorSource extends Source {
 
   /**
    * Add features without firing a `change` event.
-   * @param {Array<import("../Feature.js").default<Geometry>>} features Features.
+   * @param {Array<import("../Feature.js").default<G>>} features Features.
    * @protected
    */
   addFeaturesInternal(features) {
@@ -458,7 +458,7 @@ class VectorSource extends Source {
   }
 
   /**
-   * @param {!Collection<import("../Feature.js").default<Geometry>>} collection Collection.
+   * @param {!Collection<import("../Feature.js").default<G>>} collection Collection.
    * @private
    */
   bindFeaturesCollection_(collection) {
@@ -466,7 +466,7 @@ class VectorSource extends Source {
     this.addEventListener(
       VectorEventType.ADDFEATURE,
       /**
-       * @param {VectorSourceEvent<Geometry>} evt The vector source event
+       * @param {VectorSourceEvent<G>} evt The vector source event
        */
       function (evt) {
         if (!modifyingCollection) {
@@ -479,7 +479,7 @@ class VectorSource extends Source {
     this.addEventListener(
       VectorEventType.REMOVEFEATURE,
       /**
-       * @param {VectorSourceEvent<Geometry>} evt The vector source event
+       * @param {VectorSourceEvent<G>} evt The vector source event
        */
       function (evt) {
         if (!modifyingCollection) {
@@ -498,7 +498,7 @@ class VectorSource extends Source {
         if (!modifyingCollection) {
           modifyingCollection = true;
           this.addFeature(
-            /** @type {import("../Feature.js").default<Geometry>} */ (
+            /** @type {import("../Feature.js").default<G>} */ (
               evt.element
             )
           );
@@ -515,7 +515,7 @@ class VectorSource extends Source {
         if (!modifyingCollection) {
           modifyingCollection = true;
           this.removeFeature(
-            /** @type {import("../Feature.js").default<Geometry>} */ (
+            /** @type {import("../Feature.js").default<G>} */ (
               evt.element
             )
           );
@@ -573,7 +573,7 @@ class VectorSource extends Source {
    * stop and the function will return the same value.
    * Note: this function only iterate through the feature that have a defined geometry.
    *
-   * @param {function(import("../Feature.js").default<Geometry>): T} callback Called with each feature
+   * @param {function(import("../Feature.js").default<G>): T} callback Called with each feature
    *     on the source.  Return a truthy value to stop iteration.
    * @return {T|undefined} The return value from the last call to the callback.
    * @template T
@@ -594,7 +594,7 @@ class VectorSource extends Source {
    * value.
    *
    * @param {import("../coordinate.js").Coordinate} coordinate Coordinate.
-   * @param {function(import("../Feature.js").default<Geometry>): T} callback Called with each feature
+   * @param {function(import("../Feature.js").default<G>): T} callback Called with each feature
    *     whose goemetry contains the provided coordinate.
    * @return {T|undefined} The return value from the last call to the callback.
    * @template T
@@ -624,7 +624,7 @@ class VectorSource extends Source {
    * features, equivalent to {@link module:ol/source/Vector~VectorSource#forEachFeature #forEachFeature()}.
    *
    * @param {import("../extent.js").Extent} extent Extent.
-   * @param {function(import("../Feature.js").default<Geometry>): T} callback Called with each feature
+   * @param {function(import("../Feature.js").default<G>): T} callback Called with each feature
    *     whose bounding box intersects the provided extent.
    * @return {T|undefined} The return value from the last call to the callback.
    * @template T
@@ -647,7 +647,7 @@ class VectorSource extends Source {
    * {@link module:ol/source/Vector~VectorSource#forEachFeatureInExtent #forEachFeatureInExtent()} method instead.
    *
    * @param {import("../extent.js").Extent} extent Extent.
-   * @param {function(import("../Feature.js").default<Geometry>): T} callback Called with each feature
+   * @param {function(import("../Feature.js").default<G>): T} callback Called with each feature
    *     whose geometry intersects the provided extent.
    * @return {T|undefined} The return value from the last call to the callback.
    * @template T
@@ -657,7 +657,7 @@ class VectorSource extends Source {
     return this.forEachFeatureInExtent(
       extent,
       /**
-       * @param {import("../Feature.js").default<Geometry>} feature Feature.
+       * @param {import("../Feature.js").default<G>} feature Feature.
        * @return {T|undefined} The return value from the last call to the callback.
        */
       function (feature) {
@@ -676,7 +676,7 @@ class VectorSource extends Source {
    * Get the features collection associated with this source. Will be `null`
    * unless the source was configured with `useSpatialIndex` set to `false`, or
    * with an {@link module:ol/Collection} as `features`.
-   * @return {Collection<import("../Feature.js").default<Geometry>>} The collection of features.
+   * @return {Collection<import("../Feature.js").default<G>>} The collection of features.
    * @api
    */
   getFeaturesCollection() {
@@ -686,7 +686,7 @@ class VectorSource extends Source {
   /**
    * Get a snapshot of the features currently on the source in random order. The returned array
    * is a copy, the features are references to the features in the source.
-   * @return {Array<import("../Feature.js").default<Geometry>>} Features.
+   * @return {Array<import("../Feature.js").default<G>>} Features.
    * @api
    */
   getFeatures() {
@@ -699,7 +699,7 @@ class VectorSource extends Source {
         extend(features, getValues(this.nullGeometryFeatures_));
       }
     }
-    return /** @type {Array<import("../Feature.js").default<Geometry>>} */ (
+    return /** @type {Array<import("../Feature.js").default<G>>} */ (
       features
     );
   }
@@ -707,7 +707,7 @@ class VectorSource extends Source {
   /**
    * Get all features whose geometry intersects the provided coordinate.
    * @param {import("../coordinate.js").Coordinate} coordinate Coordinate.
-   * @return {Array<import("../Feature.js").default<Geometry>>} Features.
+   * @return {Array<import("../Feature.js").default<G>>} Features.
    * @api
    */
   getFeaturesAtCoordinate(coordinate) {
@@ -727,7 +727,7 @@ class VectorSource extends Source {
    * features.
    *
    * @param {import("../extent.js").Extent} extent Extent.
-   * @return {Array<import("../Feature.js").default<Geometry>>} Features.
+   * @return {Array<import("../Feature.js").default<G>>} Features.
    * @api
    */
   getFeaturesInExtent(extent) {
@@ -746,10 +746,10 @@ class VectorSource extends Source {
    * This method is not available when the source is configured with
    * `useSpatialIndex` set to `false`.
    * @param {import("../coordinate.js").Coordinate} coordinate Coordinate.
-   * @param {function(import("../Feature.js").default<Geometry>):boolean} [opt_filter] Feature filter function.
+   * @param {function(import("../Feature.js").default<G>):boolean} [opt_filter] Feature filter function.
    *     The filter function will receive one argument, the {@link module:ol/Feature feature}
    *     and it should return a boolean value. By default, no filtering is made.
-   * @return {import("../Feature.js").default<Geometry>} Closest feature.
+   * @return {import("../Feature.js").default<G>} Closest feature.
    * @api
    */
   getClosestFeatureToCoordinate(coordinate, opt_filter) {
@@ -770,7 +770,7 @@ class VectorSource extends Source {
     this.featuresRtree_.forEachInExtent(
       extent,
       /**
-       * @param {import("../Feature.js").default<Geometry>} feature Feature.
+       * @param {import("../Feature.js").default<G>} feature Feature.
        */
       function (feature) {
         if (filter(feature)) {
@@ -820,7 +820,7 @@ class VectorSource extends Source {
    * `source.getFeatureById(2)` will return a feature with id `'2'` or `2`.
    *
    * @param {string|number} id Feature identifier.
-   * @return {import("../Feature.js").default<Geometry>} The feature (or `null` if not found).
+   * @return {import("../Feature.js").default<G>} The feature (or `null` if not found).
    * @api
    */
   getFeatureById(id) {
@@ -832,7 +832,7 @@ class VectorSource extends Source {
    * Get a feature by its internal unique identifier (using `getUid`).
    *
    * @param {string} uid Feature identifier.
-   * @return {import("../Feature.js").default<Geometry>} The feature (or `null` if not found).
+   * @return {import("../Feature.js").default<G>} The feature (or `null` if not found).
    */
   getFeatureByUid(uid) {
     const feature = this.uidIndex_[uid];
@@ -871,7 +871,7 @@ class VectorSource extends Source {
    * @private
    */
   handleFeatureChange_(event) {
-    const feature = /** @type {import("../Feature.js").default<Geometry>} */ (
+    const feature = /** @type {import("../Feature.js").default<G>} */ (
       event.target
     );
     const featureKey = getUid(feature);
@@ -915,7 +915,7 @@ class VectorSource extends Source {
 
   /**
    * Returns true if the feature is contained within the source.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature.
+   * @param {import("../Feature.js").default<G>} feature Feature.
    * @return {boolean} Has feature.
    * @api
    */
@@ -1018,7 +1018,7 @@ class VectorSource extends Source {
    * Remove a single feature from the source.  If you want to remove all features
    * at once, use the {@link module:ol/source/Vector~VectorSource#clear #clear()} method
    * instead.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature to remove.
+   * @param {import("../Feature.js").default<G>} feature Feature to remove.
    * @api
    */
   removeFeature(feature) {
@@ -1041,8 +1041,8 @@ class VectorSource extends Source {
 
   /**
    * Remove feature without firing a `change` event.
-   * @param {import("../Feature.js").default<Geometry>} feature Feature.
-   * @return {import("../Feature.js").default<Geometry>|undefined} The removed feature
+   * @param {import("../Feature.js").default<G>} feature Feature.
+   * @return {import("../Feature.js").default<G>|undefined} The removed feature
    *     (or undefined if the feature was not found).
    * @protected
    */
@@ -1068,7 +1068,7 @@ class VectorSource extends Source {
   /**
    * Remove a feature from the id index.  Called internally when the feature id
    * may have changed.
-   * @param {import("../Feature.js").default<Geometry>} feature The feature.
+   * @param {import("../Feature.js").default<G>} feature The feature.
    * @return {boolean} Removed the feature from the index.
    * @private
    */
