diff --git a/src/ol/webgl/Buffer.js b/src/ol/webgl/Buffer.js
index de1961a85..2d77d5edd 100644
--- a/src/ol/webgl/Buffer.js
+++ b/src/ol/webgl/Buffer.js
@@ -11,9 +11,9 @@ import {assert} from '../asserts.js';
  * @enum {number}
  */
 export const BufferUsage = {
-  STATIC_DRAW: STATIC_DRAW,
-  STREAM_DRAW: STREAM_DRAW,
-  DYNAMIC_DRAW: DYNAMIC_DRAW,
+  STATIC_DRAW: 0x88e4,
+  STREAM_DRAW: 0x88e0,
+  DYNAMIC_DRAW: 0x88e8,
 };
 
 /**
