diff --git a/src/ol/xml.js b/src/ol/xml.js
index fe4b6884e..b002a2700 100644
--- a/src/ol/xml.js
+++ b/src/ol/xml.js
@@ -268,7 +268,8 @@ export function makeObjectPropertySetter(valueReader, opt_property, opt_this) {
  * @param {function(this: T, Node, V, Array<*>): void} nodeWriter Node writer.
  * @param {T} [opt_this] The object to use as `this` in `nodeWriter`.
  * @return {Serializer} Serializer.
- * @template T, V
+ * @template T
+ * @template V
  */
 export function makeChildAppender(nodeWriter, opt_this) {
   return function (node, value, objectStack) {
@@ -296,7 +297,8 @@ export function makeChildAppender(nodeWriter, opt_this) {
  * @param {function(this: T, Element, V, Array<*>): void} nodeWriter Node writer.
  * @param {T} [opt_this] The object to use as `this` in `nodeWriter`.
  * @return {Serializer} Serializer.
- * @template T, V
+ * @template T
+ * @template V
  */
 export function makeArraySerializer(nodeWriter, opt_this) {
   let serializersNS, nodeFactory;
@@ -520,7 +522,8 @@ export function serialize(
  * @param {T} [opt_this] The object to use as `this` for the node factory and
  *     serializers.
  * @return {O|undefined} Object.
- * @template O, T
+ * @template O
+ * @template T
  */
 export function pushSerializeAndPop(
   object,
