diff --git a/src/ol/xml.js b/src/ol/xml.js
index d589db53d..21089e850 100644
--- a/src/ol/xml.js
+++ b/src/ol/xml.js
@@ -266,7 +266,7 @@ export function makeObjectPropertySetter(valueReader, property, thisArg) {
  * @param {function(this: T, Node, V, Array<*>): void} nodeWriter Node writer.
  * @param {T} [thisArg] The object to use as `this` in `nodeWriter`.
  * @return {Serializer} Serializer.
- * @template T, V
+ * @template T,V
  */
 export function makeChildAppender(nodeWriter, thisArg) {
   return function (node, value, objectStack) {
@@ -294,7 +294,7 @@ export function makeChildAppender(nodeWriter, thisArg) {
  * @param {function(this: T, Element, V, Array<*>): void} nodeWriter Node writer.
  * @param {T} [thisArg] The object to use as `this` in `nodeWriter`.
  * @return {Serializer} Serializer.
- * @template T, V
+ * @template T,V
  */
 export function makeArraySerializer(nodeWriter, thisArg) {
   let serializersNS, nodeFactory;
@@ -508,7 +508,7 @@ export function serialize(
  * @param {T} [thisArg] The object to use as `this` for the node factory and
  *     serializers.
  * @return {O|undefined} Object.
- * @template O, T
+ * @template O,T
  */
 export function pushSerializeAndPop(
   object,
